store/root.js

```diff
@@ -80,6 +80,8 @@ class Root {
   @action changeActivity(url) {
     const activity =
       activities.find(({ path }) => url === path) || activities.get('overview')
+
+    if (this.activity?.path === url) return
```

Comment:
I would not add this. This is 404 and we have to handle this somehow. Maybe throwing would be better

Reply:
This is not 404. It's prevention for redirection loop. See [here](https://github.com/oacore/dashboard/blob/activities/pages/_app/index.jsx#L35)

Reply:
When does the redirection loop happen?

Reply:
Basically we change dataProvider and activity on `routeChangeComplete` which triggers autorun action and this action changes Route. Route call `routeChangeComplete` and so on.

Basically I thought MobX fired autorun only when value changed. But apparently it fires it every time value is set.